priority-channel/src/lib.rs

```diff
@@ -0,0 +1,237 @@
+// Copyright 2017-2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use std::{
+	pin::Pin,
+	result,
+	task::{Context, Poll},
+};
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+use async_channel::bounded;
+pub use async_channel::{TryRecvError, TrySendError};
+use futures::Stream;
+
+/// Create a wrapped `mpsc::broadcast` pair of `Sender` and `Receiver`.
+pub fn channel<T>(capacity: usize) -> (Sender<T>, Receiver<T>) {
+	let (tx, rx) = bounded::<T>(capacity);
+	let (tx_priority, rx_priority) = bounded::<T>(capacity);
+	(Sender { inner_priority: tx_priority, inner: tx }, Receiver { inner_priority: rx_priority, inner: rx })
+}
+
+/// A receiver tracking the messages consumed by itself.
```

Comment:
Not sure about the tracking part, can you detail this a bit? To me it looks like we just have a receiver that wraps 2 inner receivers.

Reply:
Indeed, a bad copy&paste.